config/config.go

```diff
@@ -137,6 +149,9 @@ func NewManager() *Manager {
 // Shutdown makes sure all configuration save operations are finished
 // before returning.
 func (cfg *Manager) Shutdown() {
+	if cfg == nil {
```

Comment:
I skimmed through some of the code. I did not find where a nil cfg could call Shutdown.

If that happens, if would help to not silence this error.

You could also see if you should make this method with value receiver so that Shutdown doesn't get called at all.

Reply:
True, this is not necessary.

This method could not be made a value receiver though, as this would copy the WaitGroup.